aws/resource_aws_elasticsearch_domain.go

```diff
@@ -66,6 +66,48 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
+			"advanced_security_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+							ForceNew: true,
+						},
+						"internal_user_database_enabled": {
+							Type:     schema.TypeBool,
```

Comment:
edit on second-review: let's also add the `Default: false` since we know at plan-time how it's defined by AWS (will also require a small edit to the website docs where we can simplify to just noting the default value)